extensions/exposition/source/HTTP/Server.ts

```diff
@@ -40,28 +40,17 @@ export class Server extends Connector {
       ? DEFAULTS
       : { ...DEFAULTS, ...options }
 
-    const app = express()
-
-    app.disable('x-powered-by')
-    app.use(supportedMethods(properties.methods))
-
-    return new Server(app, properties)
+    return new Server(properties)
   }
 
   public attach (process: Processing): void {
-    this.app.use((request: Request, response: Response) => {
-      const message = this.extend(request)
-
-      process(message)
-        .then(this.success(message, response))
-        .catch(this.fail(message, response))
-    })
+    this.process = process
   }
 
   protected override async open (): Promise<void> {
     const listening = promex()
 
-    this.server = this.app.listen(this.properties.port, listening.callback)
+    this.server.listen(this.properties.port, listening.callback)
 
     await listening
```

Comment:
The `promex` thing really twisted my head when I've first time seen that.
Why don't we want to use standard Node tools here?:

```ts
import { once } from 'node:events'

// then here

this.server.listen(this.properties.port);
await once(this.server, 'listening')
```

Reply:
This is not equivalent, as I understand, since `await promex` will catch the exception here.

Reply:
It's not equivalent you are right - in fact, it's better. It seems that you made an assumption that you are providing a callback function to `.listen`. In fact you are providing a function that will be added to `listening` event listeners - https://nodejs.org/api/net.html#serverlisten

So, `once` will catch an error (emitted by server) while `promex` will not.

Reply:
That may be a nice interview question: _What we will see on the output of this?_

```js
import { once } from "node:events";
import { createServer } from "node:http";

const server = createServer();
server.listen(0, "localhost-non-existent", (err, result) => {
  if (err) {
    console.log("err from callback");
  }
});
try {
  await once(server, "listening");
} catch (err) {
  console.log("err from EventEmitter");
}

```

Reply:
You're right here, I was assuming `listen` accepts old-fashion callback, like `close` does.

> Yet I believe it's not a good interview question, as it only reveals if one remembers standard APIs without documentation, which is a questionable skill.

Reply:
`.close` callback is not old fashion neither - it accepts function that would either be called with no argument or with a single error. Coincidentally it's the same as `(err, results)` in Javascript but it's not the same pattern actually.

What's lovely here that `util.promisify` do know that server expects special kind of callback and `await util.promisify(server.listen)(...)` works as expected and catches errors.